e2e/v2_curl_test.go

```diff
@@ -116,6 +116,29 @@ func TestV2CurlIssue5182(t *testing.T) {
 	}
 }
 
+func TestV2CurlIssue5246(t *testing.T) {
```

Comment:
this test does not really belong here. we should add a test in client_auth_test.go. e2e test is to ensure the correctness of etcdctl/etcd interaction, not to ensure the correctness of the server itself.